doc/test-wrapper.md

```diff
@@ -0,0 +1,123 @@
+Test Wrapper for Interactive Environments
+=========================================
+
+This document describes the usage of the `TestWrapper` submodule in the functional test suite.
+
+The TestWrapper submodule extends the `BitcoinTestFramework` functionality to external interactive environments for prototyping and educational purposes. Just like `BitcoinTestFramework`, the TestWrapper allows the user to:
+
+* Manage regtest bitcoind subprocesses.
+* Access RPC interfaces of the underlying bitcoind instances.
+* Log events to functional the test logging utility.
+
+The `TestWrapper` can be useful in interactive environments where it is necessary to extend the object lifetime of the underlying `BitcoinTestFramework` between user inputs. Such environments include the Python3 command-line interpreter or [Jupyter](https://jupyter.org/) notebooks running a Python3 kernel,
+
+## 1. Requirements
+
+* Python3
+* `bitcoind` built in the same repository as the TestWrapper.
+
+## 2. Importing TestWrapper from the Bitcoin Core repository
+
+We can import the TestWrapper by adding the path of the Bitcoin Core `test_framework` module to the beginning of the PATH variable, and then importing the `TestWrapper` class from the `test_wrapper` sub-package.
+
+```
+>>> import sys
+>>> sys.path.insert(0, "/path/to/bitcoin/test/functional")
+>>> from test_framework.test_wrapper import TestWrapper
```

Comment:
The first 2 commands didn't seem needed for running your example in the PR description?